photon-core/src/main/java/org/photonvision/common/hardware/metrics/MetricsManager.java

```diff
@@ -244,6 +244,19 @@ public double getUptime() {
 
     public void publishMetrics() {
         logger.debug("Publishing Metrics...");
+
+        // Check that the hostname hasn't changed
+        if (!metricPublisher.getTopic().getName().equals(CameraServerJNI.getHostname())) {
+            logger.warn("Metrics publisher name does not match hostname! Reinitializing publisher...");
+            metricPublisher.close();
+            metricPublisher =
```

Comment:
My only annoyance is that this code is duplicated from above. But n=2, so Big DRY can't hurt you